lumen/ai/coordinator.py

```diff
@@ -702,15 +741,115 @@ class Planner(Coordinator):
     and then executes it.
     """
 
+    planner_tools = param.List(default=[], doc="""
+        List of tools to use to provide context for the planner prior
+        to making a plan.""")
+
     prompts = param.Dict(
         default={
             "main": {
                 "template": PROMPTS_DIR / "Planner" / "main.jinja2",
                 "response_model": make_plan_models,
             },
+            "follow_up": {
+                "template": PROMPTS_DIR / "Planner" / "follow_up.jinja2",
+                "response_model": YesNo,
+            },
         }
     )
 
+    def __init__(
+        self,
+        llm: Llm | None = None,
```

Comment:
Instead of repeating all the arguments, let's disallow positional arguments on the base class.